core/src/com/projectkorra/projectkorra/airbending/AirShield.java

```diff
@@ -125,9 +126,26 @@ public void progress() {
 			this.remove();
 			return;
 		}
+		this.extinguishBlocks();
 		this.rotateShield();
 	}
 
+	private void extinguishBlocks() {
+		for (final Block testblock : GeneralMethods.getBlocksAroundPoint(this.player.getLocation(), this.radius)) {
+			if (FireAbility.isFire(testblock.getType())) {
```

Comment:
It is probably a good idea to have a `TempBlock#isTempBlock` check. So if the fire is from another ability, you remove it via the TempBlock class instead of setting the type